### PR TITLE
Feat/file type

### PR DESCRIPTION
**Pull request by jacksonwun**
Add New Question Type: File Upload with Docker Volume Support

Description:
This pull request introduces a new question type: file type. This allows users to upload files that are stored in a specified Docker volume. The implementation also supports optional validation, such as file size and format restrictions, configurable via YAML.

Key Changes:
- Added FileInput to UI 
- Added file upload type to question types.
- Implemented logic to handle file uploads and save them to a Docker volume.
- Added Max file size and file format validation.
- Defined a single Docker volume for now.
- Question format as below:
```yaml
type: file
validation:
  max_size_bytes: 5242880 # 5 * 1024 * 1024
  formats:
    - .jpg
    - .jpeg
    - .png

### CODE REVIEW

**Comment by jacksonwun** [comment]
Key Changes:
- Changed to multiplication format
- Updated README.md

```yaml
type: file
validation:
  max_size_bytes: 5*1024*1024 # 5 MB
  formats:
    - .jpg
    - .jpeg
    - .png

File: compose.yaml [hunk 1]
```diff
@@ -13,9 +13,11 @@ services:
       # - DATABASE_URL=postgres://postgres:postgres@postgres:5432/formulosity?sslmode=disable
       - DATABASE_URL=/root/sqlite3/formulosity.db
       - SURVEYS_DIR=/root/surveys
+      - UPLOADS_DIR=/root/uploads
     volumes:
       - ./api/surveys:/root/surveys
       - ./api/sqlite3:/root/sqlite3
+      - uploadvolume:/root/uploads
```
Comment (plutov) [inline compose.yaml:20]: let's keep the same format as other volumes, for example `./api/uploads` and add it to gitignore.

File: api/pkg/types/files.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package types
+
+import "io"
+
+type File struct {
+	Name string
+	Data io.Reader
+	Size int64
+	Format string
```
Comment (plutov) [inline api/pkg/types/files.go:9]: can you format this struct so types are on the same column?

File: api/pkg/types/answers.go [hunk 3]
```diff
@@ -169,3 +169,38 @@ func (a *EmailAnswer) Validate(q Question) error {
...
+func (a FileAnswer) Value() (driver.Value, error) {
+	return json.Marshal(a)
+}
+
+func (a *FileAnswer) Validate(q Question) error {
+	if q.Type != QuestionType_File && q.Validation == nil {
+		return nil
+	}
+	if q.Validation.MaxSizeBytes != nil {
+		bytes, _ := GetStringMultiplication(*q.Validation.MaxSizeBytes)
```
Comment (plutov) [inline api/pkg/types/answers.go:188]: add formatting here

File: api/pkg/storage/docker_volume.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package storage
```
Comment (plutov) [inline api/pkg/storage/docker_volume.go:1]: let's not call the file `docker_volume.go` but rather `file.go`. Also, can we add an interface and some tests?

**Review by plutov — CHANGES_REQUESTED** [review]

**Comment by plutov** [comment]
Thanks for the contribution @jacksonwun overall it looks good, just left few comments.

File: api/pkg/storage/docker_volume.go (hunk 4 above)
Comment (jacksonwun) [reply api/pkg/storage/docker_volume.go:1]: sure! will do

**Comment by jacksonwun** [comment]
Key Changes:

- Unit Tests for File answers and SaveFile Func
- Formatted bytes error message
- Trimmed to reduce max_size_bytes typo
- Improved save file func safety

**Comment by plutov** [comment]
Thanks, will test it shortly.

**Comment by plutov** [comment]
@jacksonwun it works well and I merged it. We should also make the file downloadable on the Responses page.
Screenshot 2024-10-05 at 17 34 34

**Pull request merged**